hwpf/fapi2/src/plat/plat_error.C

```diff
@@ -0,0 +1,51 @@
+#include "plat_error.H"
+#include "plat_utils.H"
+
+namespace fapi2
+{
+/*
+ * @brief Used to special handling of clock errors.
+ *
+ * The defined api is used to modify the callout and guard info
+ * specific to clock failures.
+ *
+ * @param ffdc           ffdc data which need to be updated if it is clock failure
+ * @param hwCallout      reference of hwcallout with clock related hwid
+ * @param guardRefTarget  indicates whether reference target need to be guarded or not
+ *
+ * @return void
+ */
+static void process_clock_callout(FFDC &ffdc, HWCallout& hwCallout, bool guardRefTarget)
+{
+	std::vector<uint8_t> refTargetEntityPath;
+
+	hwCallout.isPlanarCallout = true;
+	refTargetEntityPath = hwCallout.target_entity_path;
+
+	if (guardRefTarget)
+	{
+		for (auto &cdg : ffdc.hwp_errorinfo.cdg_targets)
+		{
+			if (cdg.target_entity_path == refTargetEntityPath)
```

Comment:
What should happen if the CDG and reference target are different? I think I posted a similar comment in the previous PR. - Do you have any concerns about this?

Do you need to find out the proc target from this reference target if we don't have a processor target in the CDG list?

OR

We are assuming that "Processor target will be always there in the CDG list" if so, please add some documentation on this so that it help us in the future if somebody looking at this code.

Reply:
Added the comments